### PR TITLE
Npc

### PR DESCRIPTION
**Pull request by Maxwell-Baird**
## Description

This PR creates the basic framework in which the API is going to build off of. It also includes a new table which contains a model called npc and the respected fields. The framework use Python Django as the launch point. The current issue that has spawn from this PR is formatting of the JSON so that it falls in line with the JSONAPI specs. The api currently has basic CRUD for NPC models, and can access by two routes api/v1/npcs or api/v1/npcs/{#number}.

Fixes # (issue)
Framework for the api
CRUD and database additions for npc models

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: DungeonApi/DungeonApi/settings.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'jo%8!x(b%xmmsv#n)t0=oy($3y#m^4s6(8a4nqc!t1jea&3b1j'
```
Comment (rubyr) [inline DungeonApi/DungeonApi/settings.py:23]: Would we want to add this into an .env file before we deploy?

File: DungeonApi/dungeon/views.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    elif request.method == 'POST':
+        npc_data = JSONParser().parse(request)
+        npc_serializer = NpcSerializer(data=npc_data)
+        if npc_serializer.is_valid():
+            npc_serializer.save()
+            return JsonResponse(npc_serializer.data, status=status.HTTP_201_CREATED)
+        return JsonResponse(npc_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    elif request.method == 'DELETE':
+        count = Npc.objects.all().delete()
+        return JsonResponse({'message': '{} NPCS were deleted successfully!'.format(count[0])}, status=status.HTTP_204_NO_CONTENT)
```
Comment (rubyr) [inline DungeonApi/dungeon/views.py:28]: Would we want to add authentication to this eventually?

Comment (Maxwell-Baird) [reply DungeonApi/dungeon/views.py:28]: This was more for practice for us to get use to how django does api calls, but if in the future we want to use this calls then yes we will need to add authentication

File: DungeonApi/DungeonApi/settings.py (hunk 1 above)
Comment (Maxwell-Baird) [reply DungeonApi/DungeonApi/settings.py:23]: Thanks for reminding about including the env

**Pull request merged**